engine/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/AdvisorEmailAggregator.java

```diff
@@ -0,0 +1,85 @@
+    private static final String DEACTIVATED_RECOMMENDATION = "resolved_recommendation";
+
+    // Advisor events aggregator
+    private static final String ADVISOR_KEY = "advisor";
+    private static final String NEW_RECOMMENDATIONS = "new_recommendations";
+    private static final String RESOLVED_RECOMMENDATIONS = "resolved_recommendations";
+    private static final String DEACTIVATED_RECOMMENDATIONS = "deactivated_recommendations";
+
+    // Advisor event types
+    private static final String NEW_RECOMMENDATION_EVENT = "new-recommendation";
+    private static final String RESOLVED_RECOMMENDATION_EVENT = "resolved-recommendation";
+    private static final String DEACTIVATED_RULE_EVENT = "deactivated-rule";
+    private static final Set<String> EVENT_TYPES = new HashSet<>(Arrays.asList(
+        NEW_RECOMMENDATION_EVENT, RESOLVED_RECOMMENDATION_EVENT,
+        DEACTIVATED_RULE_EVENT
+    ));
+
+    // Advisor final payload helpers
+    private final HashMap<String, HashSet<String>> newRecommendations = new HashMap<String, HashSet<>>();
+    private final HashMap<String, HashSet<String>> resolvedRecommendations = new HashSet<String, HashSet<>>();
+    private final HashSet<String> deactivatedRecommendations = new HashSet<>();
+
+    public AdvisorDailyDigestEmailPayloadAggregator() {
+        JsonObject advisor = new JsonObject();
+
+        context.put(ADVISOR_KEY, advisor);
+    }
+
+    @Override
+    void processEmailAggregation(EmailAggregation notification) {
+        JsonObject notifPayload = notification.getPayload();
+        JsonObject advisor = context.getJsonObject(ADVISOR_KEY);
+        String eventType = notifPayload.getString(EVENT_TYPE_KEY);
+        String inventoryId = notification.getContext().getString(INVENTORY_ID);
+
+        if (!EVENT_TYPES.contains(eventType)) {
+            return;
+        }
+
+        notifPayload.getJsonArray(EVENTS_KEY).stream().forEach(eventObject -> {
+            JsonObject event = (JsonObject) eventObject;
+            JsonObject payload = event.getJsonObject(PAYLOAD_KEY);
+            String ruleId = payload.getString(PAYLOAD_RULE_ID);
+
+            if (eventType == NEW_RECOMMENDATION) {
+                if (! newRecommendations.contains(ruleId)) {
+                    newRecommendations.put(ruleId, new HashSet<String>());
+                }
+                newRecommendations.get(ruleId).add(inventoryId);
+            } else if (eventType == RESOLVED_RECOMMENDATION) {
+                if (! resolvedRecommendations.contains(ruleId)) {
+                    resolvedRecommendations.put(ruleId, new HashSet<String>());
+                }
+                resolvedRecommendations.get(ruleId).add(inventoryId);
+            } else if (eventType == DEACTIVATED_RECOMMENDATION) {
+                deactivatedRecommendations.add(ruleId);
+            }
+        });
+        advisor.put(NEW_RECOMMENDATIONS, newRecommendations);
```

Comment:
Each invocation of the `processEmailAggregation` method relies on the same `advisor` instance. This means you don't have to do this `put` and the others below on each invocation, but only once from the class constructor (`public AdvisorDailyDigestEmailPayloadAggregator()`).

Reply:
Yeah, I was wondering about that.  So I need to basically get those in the class constructor and then put data into them in `processEmailAggregation`, yeah?